utils/api/definitions.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2018 Caicloud Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"encoding/json"
+
+	"github.com/caicloud/nirvana/definition"
+	"github.com/caicloud/nirvana/service"
+)
+
+// Parameter describes a function parameter.
+type Parameter struct {
+	// Source is the parameter value generated from.
+	Source definition.Source
+	// Name is the name to get value from a request.
+	Name string
+	// Description describes the parameter.
+	Description string
+	// Type is parameter object type.
+	Type TypeName
+	// Default is encoded default value.
+	Default []byte
+}
+
+// Result describes a function result.
+type Result struct {
+	// Destination is the target for the result.
+	Destination definition.Destination
+	// Description describes the result.
+	Description string
+	// Type is result object type.
+	Type TypeName
+}
+
+// Example is just an example.
+type Example struct {
+	// Description describes the example.
+	Description string
+	// Type is result object type.
+	Type TypeName
+	// Instance is encoded instance data.
+	Instance []byte
+}
+
+// Definition is complete version of def.Definition.
+type Definition struct {
```

Comment:
why do you redefine everything here again?.... this is not maintainable

Reply:
I'll refactor this package when I'm writing code generation.